lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/sys.rb

```diff
@@ -302,6 +319,8 @@ def cmd_shell(*args)
         end
 
         return true
+      when '-i'
+        raw = true
```

Comment:
TODO: We'll want to add validation for ensuring that raw mode needs to be coupled with a pty

Reply:
I'm probably missing it; Was there validation added for this? 👀

Reply:
yup! the raw flag is now only added when the `pty_shell` function is called
https://github.com/rapid7/metasploit-framework/pull/15522/files/d98896564d25ebdfe5671d637647b14327aa2956#diff-f6355bf5fe7f8b6713ddc4f22831b4d74290abb1296dccce03289b88be417226R383